tutorials/io/binary_serialization_api.rst

```diff
@@ -383,13 +393,69 @@ For each Name and Sub-Name
 Every name string is padded to 4 bytes.
 
 16: :ref:`RID<class_rid>` (unsupported)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-17: :ref:`Object<class_object>` (unsupported)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+17: :ref:`Object<class_object>`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An Object could be serialized in three different ways: as a null value, with
+``full_objects = false``, or with ``full_objects = true``.
+
+A null value
+""""""""""""
+
++----------+-------+------------+-------------------------------------------------+
+| Offset   | Len   | Type       | Description                                     |
++==========+=======+============+=================================================+
+| 4        | 4     | Integer    | Zero (32-bit signed integer)                    |
++----------+-------+------------+-------------------------------------------------+
+
+``full_objects`` disabled
+"""""""""""""""""""""""""
+
++----------+-------+------------+-------------------------------------------------+
+| Offset   | Len   | Type       | Description                                     |
++==========+=======+============+=================================================+
+| 4        | 8     | Integer    | The Object instance ID (64-bit signed integer)  |
++----------+-------+------------+-------------------------------------------------+
+
+``full_objects`` enabled
+""""""""""""""""""""""""
+
++----------+-------+------------+-------------------------------------------------+
+| Offset   | Len   | Type       | Description                                     |
++==========+=======+============+=================================================+
+| 4        | X     | Bytes      | Class name (UTF-8 encoded string)               |
```

Comment:
Good point! I tried writing something that I think makes sense for this.

Reply:
Actually now that I've looked at it I think my suggestion was wrong: position/type 4 listed in this exact file refers to how String is stored as a separate Variant. So there length is being at offset 4 because before it goes (type+flags). Here type and flags aren't stored for this string, it's just length + actual UTF-8 encoded string. So length doesn't have additional 4 bytes offset here and pointing to the type 4 in the list might be misleading. Take a look at how strings are listed in `PackedStringArray` (type 25), here it should be listed similarly. Sorry for that.
So probably something like this:
```
``full_objects`` enabled
""""""""""""""""""""""""

+----------+-------+----------------+----------------------------------------------------------+
| Offset   | Len   | Type           | Description                                              |
+==========+=======+================+==========================================================+
| 4        | 4     | Integer        | Class name (String length)                               |
+----------+-------+----------------+----------------------------------------------------------+
| 8        | X     | Bytes          | Class name (UTF-8 encoded string)                        |
+----------+-------+----------------+----------------------------------------------------------+
| X+8      | 4     | Integer        | The number of properties that are serialized             |
+----------+-------+----------------+----------------------------------------------------------+

For each property:

+----------+-------+----------------+----------------------------------------------------------+
| Offset   | Len   | Type           | Description                                              |
+==========+=======+================+==========================================================+
| Y        | 4     | Integer        | Property name (String length)                            |
+----------+-------+----------------+----------------------------------------------------------+
| Y+4      | Z     | Bytes          | Property name (UTF-8 encoded string)                     |
+----------+-------+----------------+----------------------------------------------------------+
| Y+4+Z    | W     | <variable>     | Property value, using this same format                   |
+----------+-------+----------------+----------------------------------------------------------+
```
Other than this (and [this](https://github.com/godotengine/godot/pull/49470#discussion_r650119429)) great job!

Reply:
Yeah, that makes sense. Ok, done!